### PR TITLE
Set location for a single simulator

### PR DESCRIPTION
**Pull request by valheru7**
This leaves default behavior of 'all booted simulators' intact but adds an additional option to only change the location for one simulator.

### CODE REVIEW

**Comment by valheru7** [comment]
PTAL @keith

File: sources/simulators.swift [hunk 1]
```diff
@@ -6,7 +6,37 @@ enum SimulatorFetchError: String, Error {
...
+
+    var udid: String
+    var name: String
+    var state: State
+}
+
+func getDeviceUdid(name: String) throws -> Result<String> {
+    let simulators: [Simulator] = try getSimulators().filter { $0.name == name }
+
+    guard let simulator = simulators.first else {
```
Comment (kastiglione) [inline sources/simulators.swift]: Instead of `filter` + `first`, could use `first(where:)`.

File: sources/simulators.swift [hunk 2]
```diff
@@ -29,14 +59,21 @@ func getBootedSimulators() throws -> [String] {
...
-    let bootedIDs = devices
-        .flatMap { $1 }
-        .filter { $0["state"] == "Booted" }
-        .flatMap { $0["udid"] }
 
-    if bootedIDs.isEmpty {
-        throw SimulatorFetchError.noBootedSimulators
+    var simulators = [Simulator]()
+    for topName in devices.keys {
+        if let deviceTypes = devices[topName] {
```
Comment (kastiglione) [inline sources/simulators.swift]: Can this be `for (topName, deviceTypes) in devices`?

Comment (valheru7) [reply sources/simulators.swift]: yeah much cleaner. thanks =)

**Comment by keith** [comment]
Worked with Jerica offline and came up with a slightly different approach https://github.com/lyft/set-simulator-location/pull/7

**Pull request closed without merging**